build.sbt

```diff
@@ -8,7 +8,7 @@ scalaVersion in ThisBuild := "2.11.8"
 
 scalacOptions ++= Seq("-unchecked", "-optimise", "-deprecation",
   "-Xcheckinit", "-encoding", "utf8", "-feature", "-Yinline-warnings",
-  "-Xfatal-warnings"
+  "-Xfatal-warnings", "-Ybackend:GenBCode"
```

Comment:
what is this for?

Reply:
See the commit description. In short I hit an optimiser bug that is solved by switching to an alternate (newer) optimiser that is default in 2.12.